src/xxh3.rs

```diff
@@ -459,10 +458,10 @@ fn hash64_0(secret: &[u8], seed: u64) -> u64 {
 }
 
 fn hash64_1to3(data: &[u8], secret: &[u8], seed: u64) -> u64 {
-    let x1 = data[0] as u32;
-    let x2 = data[data.len() >> 1] as u32;
-    let x3 = *data.last().unwrap() as u32;
-    let x4 = data.len() as u32;
+    let x1: u32 = data[0].try_into().unwrap();
+    let x2: u32 = data[data.len() >> 1].try_into().unwrap();
+    let x3: u32 = (*data.last().unwrap()).try_into().unwrap();
+    let x4: u32 = data.len().try_into().unwrap();
```

Comment:
This should be a truncating cast. Otherwise it will fail on inputs > 2^32. The reference implementation truncates: https://github.com/Cyan4973/xxHash/blob/dev/xxhash.h#L3845